### DISCUSSION TITLE
Using added EBS volumes

### DISCUSSION THREAD

**Discussion by abhijna** [nih-cfde/training-and-engagement#413] [AWS] [answered]
How to make an EBS volume available for use in Linux?

**Comment by abhijna** [comment, marked as answer]
# How to connect to an added storage volume

- Add an additional storage volume and launch your instance
- Connect to it using the web terminal or ssh
- Run `lsblk` to list information about all available or the specified block devices. 
```
lsblk
```
The lsblk command reads the sysfs filesystem and udev db to gather information. If the udev db is not available or lsblk is compiled without udev support, then it tries to read LABELs, UUIDs and filesystem types from the block device.
- The output should look something like this:

```
ubuntu@user-ip~$ lsblk
NAME    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
loop0     7:0    0 33.3M  1 loop /snap/amazon-ssm-agent/3552
loop1     7:1    0 55.5M  1 loop /snap/core18/1997
loop2     7:2    0 70.4M  1 loop /snap/lxd/19647
loop3     7:3    0 32.3M  1 loop /snap/snapd/11588
xvda    202:0    0    8G  0 disk 
└─xvda1 202:1    0    8G  0 part /
xvdb    202:16   0    8G  0 disk 
```


- `/dev/xvda1` is the root volume, and `/dev/xvdb` is the new disk

The new disc is not mounted yet.

## How to mount your volume

### Create a new file system

- New volumes are raw block devices, and you must create a file system on them before you can mount and use them. 

Please note!
Volumes that were created from snapshots likely have a file system on them already; if you create a new file system on top of an existing file system, the operation overwrites your data.


- Use the ` sudo file -s` command to get information about a specific device, such as its file system type. If the output shows simply data, as in the following example output, there is no file system on the device
If you are trying to mount a volume to an instance that you just set up, it probably does not have a file system and your output will look like this:

Your code:
```
sudo file -s /dev/xvdb
```

Expected output:
```
/dev/xvdb: data
```

- Add a file system by typing:
```
sudo mkfs -t xfs /dev/xvdb
```

## Mount the volume

- Use the mkdir command to create a mount point directory for the volume. The mount point is where the volume is located in the file system tree and where you read and write files to after you mount the volume. The following example creates a directory named /data.

```
sudo mkdir /data
```


- Use the following command to mount the volume at the directory you created in the previous step.

```
sudo mount /dev/xvdb /data
```

Please note!
The mount point is not automatically preserved after rebooting your instance. 

[Follow these instructions to automatically mount upon reboot](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-using-volumes.html#ebs-mount-after-reboot)

## To work in the newly mounted volume:
- First, double-check if your volume is mounted and has a file system by typing `lsblk`

Your output should look like this:

```
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
loop0         7:0    0 55.5M  1 loop /snap/core18/1997
loop1         7:1    0 33.3M  1 loop /snap/amazon-ssm-agent/3552
loop2         7:2    0 70.4M  1 loop /snap/lxd/19647
loop3         7:3    0 32.3M  1 loop /snap/snapd/11588
xvdb    259:0    0   8G  0 disk /data
xvda    202:0    0    8G  0 disk 
└─xvda1 202:1    0    8G  0 part /
```

The MOUNTPOINT column shows that the xvdb volume is mounted with the /data file system. 

- You can  `cd` into the volume by typing:
```
cd /data
```
- You can now install and run programs and save data in this newly mounted volume.

**Reply by abhijna** [reply]
@marisalim and @s-canchi can one of you please test this tutorial if you have some time next week?

**Reply by luizirber** [reply]
> * Move into the dev folder from the home directory like so:
> 
> ```
> cd ../../dev
> ```

You don't need to move into the `/dev` folder to use `lsblk` (and it might be confusing to move around, in a tutorial context =])
